### PR TITLE
added algorithms by languages section

### LINKED ISSUE #759: 💡 [FEATURE] - Add DSA Projects
### Idea Contribution

- [X] I have read all the feature request issues.
- [X] I'm interested in working on this issue
- [X] I'm part of GSSOC organization

### Explain feature request

DSA is most important skills were every computer science student should know. Creating projects will provide better  understanding in Data Structures.

### Explain your solution

I want to add "Projects" section in DSA, I will add tutorials of following projects
1. Snake Game
2. Sudoko Solver
3. Sorting Visualizer
4. File Zipper

### Any alternative approaches/features

_No response_

### Additional Context

_No response_

### PR DESCRIPTION
**Pull request by ImmidiSivani**
## Description
added algorithms for 3 languages in DSA .

Closes #759 

## Type of change
- [x] Adding a new Resource
- [x] Adding a new Category
- [ ] Documentation enhancement or fixes
- [ ] Website enhancement or fixes

## Checklist:
- [x] I read carefully [CONTRIBUTING.md](https://github.com/jfmartinz/ResourceHub/blob/main/CONTRIBUTING.md)
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation

## Screenshots (if appropriate):

Please add screenshots to help explain your changes.

### CODE REVIEW

File: Data Structures and Algorithm (DSA)/readme.md [hunk 1]
```diff
@@ -218,6 +219,55 @@ To the Point, Short and Practical - These 3 terms defines our Content
 </tr>
 </table>
 
+### Algorithms by Languages
+
```
Comment (jfmartinz) [inline Data Structures and Algorithm (DSA)/readme.md:223]:
```suggestion
Add a description here
```

File: Data Structures and Algorithm (DSA)/readme.md [hunk 2]
```diff
@@ -218,6 +219,55 @@ To the Point, Short and Practical - These 3 terms defines our Content
...
+</tr>
+<tr>
+<td><a href="https://www.youtube.com/watch?v=cVZMah9kEjI&t=1s">Merge Sort</a></td>
+</tr>
+<tr>
+<td><a href="https://www.youtube.com/watch?v=cVZMah9kEjI&t=1s">Binary Search</a></td>
+</tr>
+
+
+</table>
```
Comment (jfmartinz) [inline Data Structures and Algorithm (DSA)/readme.md:269]: You forgot to add a description in each resource. Check this [link](https://github.com/jfmartinz/ResourceHub/blob/main/CONTRIBUTING.md#how-to-add-a-resource)

**Review by jfmartinz — CHANGES_REQUESTED** [review]

**Review by jfmartinz — CHANGES_REQUESTED** [review]
Each language should have its own table like this:

### JavaScript
### Section name
<table width="100%">
 <tr>
   <th>Resource Name</th>
   <th>Description</th>
 </tr>
 <tr>
   <td><a href="URL HERE">RESOURCE NAME</a></td>
   <td>RESOURCE DESCRIPTION</td>
 </tr>
</table>

Java
### Section name
<table width="100%">
 <tr>
   <th>Resource Name</th>
   <th>Description</th>
 </tr>
 <tr>
   <td><a href="URL HERE">RESOURCE NAME</a></td>
   <td>RESOURCE DESCRIPTION</td>
 </tr>
</table>

I hope this is clear

**Comment by jfmartinz** [comment]
You just need to add '>' before the description, why did  you close? After that change I can merged it

**Comment by ImmidiSivani** [comment]
This branch has conflicts shall I create new branch and open new PR for review,if it is okay I will close it.

**Comment by jfmartinz** [comment]
You can fix it if you want [here](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/addressing-merge-conflicts/resolving-a-merge-conflict-using-the-command-line)

**Pull request closed without merging**